Kafka, MongoDB, Redis: Publish templated connectors (#324)
mulesoft/docs-connectors: modules/ROOT/pages/kafka/kafka-connector-examples.adoc

= Kafka Examples - Mule 4
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../../assets/images/

== Consume a Topic

> Kafka connector can be used for consuming messages from a particular topic and feeding the flow with those messages, or producing a message into a topic. This example uses the two Kafka source operations, *Message Consumer* and *Publish Message* to push a message to Kafka and retrieve it.

Reviewer: ```suggestion
You can use Kafka connector  to consume messages from a particular topic and then feed the flow with those messages, or to produce a message into a topic. This example uses the two Kafka source operations, *Message Consumer* and *Publish Message*, to push a message to Kafka and then retrieve it.
```

|===
+
. Add the values for each field in the `mule-app.properties` file.
. Drag *Logger* to the canvas and set the message to:
+
[source,xml]
----
#['New message arrived: ' ++ payload ++ &quot;, key:&quot; ++ attributes.key ++ &quot;, partition:&quot; ++ attributes.partition ++ &quot;, offset:&quot; ++ attributes.offset ]
----
+
The Consumer flow appears in Studio as:
+
image::kafka/kafka-consumer-studio-flow.png[]
+
. In the next flow, drag *HTTP Listener* to the canvas, set the *Path* to `/pushMessage`, and use the defaults for the configuration.
. Drag *Logger* to the canvas and set the message to:
+
[source,xml]
----
#[&quot;Message: '&quot; ++ payload.message ++ &quot;' is going to be published to topic: '&quot; ++ payload.topic ++ &quot;'.&quot;]
----
+
. Drag the Kafka *Publish Message* operation to the Studio canvas, set *Topic* to `#[payload.topic]`, and set *Key* to `#[now()]`.
. Drag *Set Payload* to the canvas and set the *Value* to `Message successfully sent to Kafka topic.`
+
The Producer flow appears in Studio as:
+
image::kafka/kafka-producer-studio-flow.png[]

> === XML for Consume a Topic

Reviewer: ```suggestion
=== XML for Consuming a Topic
```
